src/sentry/static/sentry/app/components/gridEditable/index.tsx

```diff
@@ -141,86 +141,77 @@ class GridEditable<
     this.setState({isEditing: nextValue});
   };
 
-  openModalAddColumnAt = (insertIndex: number) => {
+  /**
+   * Leave `insertIndex` as undefined to add new column to the end.
+   */
+  openModalAddColumnAt = (insertIndex: number = -1) => {
+    if (insertIndex < 0) {
+      insertIndex = this.props.columnOrder.length;
+    }
+
     return this.toggleModalEditColumn(insertIndex);
   };
 
   toggleModalEditColumn = (
     indexColumnOrder?: number,
     column?: GridColumn<ColumnKey>
   ): void => {
-    if (this.state.isEditing) {
-      const {modalEditColumn} = this.props;
-
-      openModal(openModalProps => (
-        <GridModalEditColumn
-          {...openModalProps}
-          indexColumnOrder={indexColumnOrder}
-          column={column}
-          renderBodyWithForm={modalEditColumn.renderBodyWithForm}
-          renderFooter={modalEditColumn.renderFooter}
-        />
-      ));
-    }
+    const {modalEditColumn} = this.props;
+
+    openModal(openModalProps => (
+      <GridModalEditColumn
+        {...openModalProps}
+        indexColumnOrder={indexColumnOrder}
+        column={column}
+        renderBodyWithForm={modalEditColumn.renderBodyWithForm}
+        renderFooter={modalEditColumn.renderFooter}
+      />
+    ));
   };
 
-  renderError = () => {
-    const {error} = this.props;
+  renderHeaderButton = () => {
+    if (!this.props.isEditable) {
+      return null;
+    }
 
     return (
-      <React.Fragment>
-        <GridPanel>
-          <Grid
-            isEditable={this.props.isEditable}
-            isEditing={this.state.isEditing}
-            numColumn={this.state.numColumn}
-          >
-            {this.renderGridHead()}
-            <GridBody>
-              <GridRow>
-                <GridBodyCellSpan>
-                  <GridBodyErrorAlert type="error" icon="icon-circle-exclamation">
-                    {error}
-                  </GridBodyErrorAlert>
-                </GridBodyCellSpan>
-              </GridRow>
-            </GridBody>
-          </Grid>
-        </GridPanel>
-      </React.Fragment>
+      <HeaderButton onClick={() => this.openModalAddColumnAt()}>
+        <InlineSvg src="icon-circle-add" />
+        {t('Add Column')}
+      </HeaderButton>
     );
   };
 
-  renderLoading = () => {
-    return (
-      <GridBody>
-        <GridRow>
-          <GridBodyCellSpan>
-            <GridBodyCellLoading>
-              <LoadingContainer isLoading />
-            </GridBodyCellLoading>
-          </GridBodyCellSpan>
-        </GridRow>
-      </GridBody>
-    );
-  };
+  renderGridHeadEditButtons = () => {
+    if (!this.props.isEditable) {
+      return null;
+    }
+
+    const sorryMessage = t('This will be improved soon! Please bear with us!');
```

Comment:
Are you planning to do follow up pull requests to remove the need to apologize?

Reply:
There was a chance that the table might be in a limbo state for a while but it's not necessary now.